velox/exec/tests/WindowTest.cpp

```diff
@@ -487,5 +487,31 @@ TEST_F(WindowTest, nagativeFrameArg) {
   }
 }
 
+TEST_F(WindowTest, frameColumnNullCheck) {
+  auto input = {makeRowVector(
+      {"c0", "p0", "s0", "off0"},
+      {
+          makeNullableFlatVector<int64_t>({1, std::nullopt, 1, 2, 2}),
+          makeFlatVector<int64_t>({1, 2, 1, 2, 1}),
+          makeNullableFlatVector<int64_t>({1, 2, 3, std::nullopt, 5}),
+          makeNullableFlatVector<int64_t>({2, std::nullopt, 4, 5, 6}),
+      })};
+
+  auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+  CursorParameters params;
+  auto queryCtx = core::QueryCtx::create(executor_.get());
+  params.queryCtx = queryCtx;
+  params.planNode =
+      PlanBuilder(planNodeIdGenerator)
+          .values(input)
+          .window(
+              {"sum(c0) OVER (PARTITION BY p0 ORDER BY s0 RANGE BETWEEN UNBOUNDED PRECEDING AND off0 FOLLOWING)"})
+          .planNode();
+
+  VELOX_ASSERT_THROW(
+      readCursor(params, [](Task*) {}),
+      "Frame column should be NULL iff ORDER BY value is NULL");
+}
```

Comment:
Could we add another test case that logically shouldn't throw but throw with the original code? E.g.,
```
auto input = {makeRowVector(
      {"c0", "p0", "s0", "off0"},
      {
          makeNullableFlatVector<int64_t>({1, 1, 2, std::nullopt, 2}),
          makeFlatVector<int64_t>({1, 1, 1, 1, 1}),
          makeNullableFlatVector<int64_t>({1, std::nullopt, 2, 3, 5}),
          makeNullableFlatVector<int64_t>({2, std::nullopt, 4, 5, 6}),
      })};

  auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
  auto plan =
      PlanBuilder(planNodeIdGenerator)
          .values(input)
          .window(
              {"sum(c0) OVER (PARTITION BY p0 ORDER BY s0 RANGE BETWEEN UNBOUNDED PRECEDING AND off0 FOLLOWING)"})
          .planNode();
  ASSERT_NO_THROW(AssertQueryBuilder(plan).copyResults(pool()));
```